src/v/redpanda/admin/server.cc

```diff
@@ -708,7 +708,7 @@ void admin_server::log_level_timer_handler() {
     rearm_log_level_timer();
 }
 
-ss::future<ss::httpd::redirect_exception> admin_server::redirect_to_leader(
```

Comment:
the only downside i can see is that if it were futurized in the future (maybe it was in the past?) that afaik `throw` will happily throw a future so updating all the callers would need to be robustly changed.

Reply:
> if it were futurized in the future

Well, this applies to many other scenarios. E.g. I once accidentally futurized an argument of `ssx::async_for_each` and it at the time it silently swallowed it and even mostly worked. Also, I guess we'll figure out it's broken pretty quickly.

> maybe it was in the past?

Surprisingly it never was, I checked whole git history of the file! Do you think it was a "defensive futurization" to avoid the scenario of futurizing it in future and throwing a future?

Reply:
> Surprisingly it never was, I checked whole git history of the file! Do you think it was a "defensive futurization" to avoid the scenario of futurizing it in future and throwing a future?

No, I don't think it was defensive. I think the author probably believed it was going to need to become futurized for some other reason in the near future but it never did.

Seems like a reasonable change.